### PR TITLE
file based logging

### PR DESCRIPTION
**Pull request by sarob**
- implements /log/app.log
- app_config.py has new vars
  - LOG_TO_FILE
  - LOG_TO_CONSOLE
-  thread pooling fixes to support file logging
- to support the PR #555

### CODE REVIEW

File: src/aihawk_easy_applier.py [hunk 1]
```diff
@@ -21,6 +21,10 @@
 
 import src.utils as utils
 from loguru import logger
+import threading
+
+# Create a thread-local storage
+local = threading.local()
```
Comment (surapuramakhil) [inline src/aihawk_easy_applier.py:27]: why is thread local required? here

File: src/utils.py [hunk 2]
```diff
@@ -1,24 +1,68 @@
...
+# Remove default logger
+logger.remove()
+
+# Configure Loguru logger
+config = {
+    "handlers": []
+}
+
+# Add file logger if LOG_TO_FILE is True
+if LOG_TO_FILE:
```
Comment (surapuramakhil) [inline src/utils.py:26]: can you use simple logging handler? it's not wise to reinvent the well.

Python do profile FileHandler, BaseRotatingHandler and other things.

Comment (surapuramakhil) [reply src/utils.py:26]: and remove multithreading, as it is adding complexity which is completely unnecessary.

**Comment by surapuramakhil** [comment]
when logging is concerned - we need to have module-based logging. As of now we used same logger entire code base, as the result we can't set log level at module levels which increases debugging time and other lot of impacts.

File: src/aihawk_easy_applier.py (hunk 1 above)
Comment (feder-cr) [reply src/aihawk_easy_applier.py:27]: yes, why?

Comment (sarob) [reply src/aihawk_easy_applier.py:27]: I was solving for some concurrency errors that seem to have disappeared.

updated with new push.

**Comment by surapuramakhil** [comment]
Hi @sarob can you separate logging from utils.py. like having logging.py which can take care of configuring and managing logger

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
Everything is good. Only 1 thing is required. we can move logging configuration from utils.py to separate file logging.py

**Comment by sarob** [comment]
Will do tomorrow morning. Sean ***@***.*** On Oct 26, 2024, at 20:01, Akhil ***@***.***> wrote:﻿
@surapuramakhil requested changes on this pull request.

Everything is good. Only 1 thing is required. we can move logging configuration from utils.py to separate file logging.py

**Comment by sarob** [comment]
I need @feder-cr to resolve the conflict while repo rights and roles are being worked out. 

I have the suggested changes to move log functions to logging.py ready to push.

**Review by surapuramakhil — APPROVED** [review]

**Comment by sarob** [comment]
tests pass and double checked functionality

**Pull request merged**